### PR TITLE
drivers/pinmux: stm32: Add i2c4 pinmux to stm32f7.h

### PR DESCRIPTION
**Pull request by DenvE**
Add all i2c4 gpio mapping to pinmux to enable i2c4 on stm32f7 based boards

### CODE REVIEW

**Comment by jukkar** [comment]
This looks like a duplicate of #21834, please do not create new PRs for each push. Just force push your branch to existing PR. You need to decide which PR you want to work with and then stick to it.

File: subsys/net/lib/sockets/sockets_tls.c [hunk 1]
```diff
@@ -1797,7 +1797,7 @@ static int ztls_poll_update_ctx(struct net_context *ctx,
 			/* EAGAIN might happen during or just after
 			 * DTLS handshake.
 			 */
-			if (recv(pfd->fd, NULL, 0, ZSOCK_MSG_DONTWAIT) < 0 &&
+			if (zsock_recv(pfd->fd, NULL, 0, ZSOCK_MSG_DONTWAIT) < 0 &&
```
Comment (pfalcon) [inline subsys/net/lib/sockets/sockets_tls.c]: This commit should not be part of the PR titled "drivers/pinmux: stm32: Add i2c4 pinmux to stm32f7.h". If it's really needed, please submit it as a separate PR.

**Review by pfalcon — CHANGES_REQUESTED** [review]
Commits unrelated to the PR topic.

**Comment by DenvE** [comment]
Hi Paul,

I have followed the instruction by
Robert Lubos

@DenvE <https://github.com/DenvE> There's no need to add extra complexity
with #if defined. Simply replacing the existing recv call with zsock_recv is
a safe operation - they're equivalent and the latter is always available.
recv is more commonly used alias for zsock_recv, though it needs extra
configs to be enabled (as it's a part of POSIX API).

IMO it's would be acceptable to have this change as well in this PR, as a
separate commit.


Br,


Wim

**Comment by pfalcon** [comment]
@DenvE: Each of reviewers express their opinion. My opinion is simple and clear - there should be no stuff like socket changes hidden in pull requests titled "drivers/pinmux: stm32: Add i2c4 pinmux to stm32f7.h". This is both very simple (to resolve/avoid) and serious problem - we'll have neither adequate review process nor resulting project code quality is people will hide unrelated changes within other changes. (And yes, I humbly consider this "hiding". If that's not the intention, there's again an easy way to fix that by submitting unrelated changes in a separate PR. Otherwise, YMMV.)

**Comment by DenvE** [comment]
Hi Paul,

I do agree.

kr,
Wim

**Comment by jukkar** [comment]
Already fixed by #22359
